### PR TITLE
migrate some cupertino files to nullsafety

### PR DESCRIPTION
**Pull request by a14n**
## Description

NNBD migration for part of cupertino

## Related Issues

* dart-lang/sdk#40146

## Tests

I added the following tests:

None

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/context_menu.dart [hunk 1]
```diff
@@ -776,7 +774,7 @@ class _ContextMenuRoute<T> extends PopupRoute<T> {
     // animate the entire page into the scene. In the case of _ContextMenuRoute,
     // two individual pieces of the page are animated into the scene in
     // buildTransitions, and null is returned here.
-    return null;
+    return Container();
```
Comment (goderbauer) [inline packages/flutter/lib/src/cupertino/context_menu.dart:777]: The comment above needs updating.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by virakpanha** [comment]
@virakpanha

File: packages/flutter/lib/src/cupertino/action_sheet.dart [hunk 2]
```diff
@@ -225,12 +223,12 @@ class CupertinoActionSheet extends StatelessWidget {
       if (cancelButton != null) _buildCancelButton(),
     ];
 
-    final Orientation orientation = MediaQuery.of(context).orientation;
+    final Orientation orientation = MediaQuery.of(context)!.orientation;
```
[the remark this answers is not in the document]
Comment (a14n) [reply packages/flutter/lib/src/cupertino/action_sheet.dart:228]: Done

File: packages/flutter/lib/src/cupertino/action_sheet.dart [hunk 3]
```diff
@@ -399,15 +397,15 @@ class _CupertinoAlertRenderWidget extends RenderObjectWidget {
   @override
   RenderObject createRenderObject(BuildContext context) {
     return _RenderCupertinoAlert(
-      dividerThickness: _kDividerThickness / MediaQuery.of(context).devicePixelRatio,
-      dividerColor: CupertinoDynamicColor.resolve(_kButtonDividerColor, context),
+      dividerThickness: _kDividerThickness / MediaQuery.of(context)!.devicePixelRatio,
```
[the remark this answers is not in the document]
Comment (a14n) [reply packages/flutter/lib/src/cupertino/action_sheet.dart:403]: Done

File: packages/flutter/lib/src/cupertino/bottom_tab_bar.dart [hunk 4]
```diff
@@ -139,14 +137,14 @@ class CupertinoTabBar extends StatelessWidget implements PreferredSizeWidget {
...
     final Color backgroundColor =
         this.backgroundColor ?? CupertinoTheme.of(context).barBackgroundColor;
-    return CupertinoDynamicColor.resolve(backgroundColor, context).alpha == 0xFF;
+    return CupertinoDynamicColor.resolve(backgroundColor, context)!.alpha == 0xFF;
   }
 
   @override
   Widget build(BuildContext context) {
-    final double bottomPadding = MediaQuery.of(context).padding.bottom;
+    final double bottomPadding = MediaQuery.of(context)!.padding.bottom;
```
[the remark this answers is not in the document]
Comment (a14n) [reply packages/flutter/lib/src/cupertino/bottom_tab_bar.dart:146]: Done

File: packages/flutter/lib/src/cupertino/context_menu.dart (hunk 1 above)
Comment (a14n) [reply packages/flutter/lib/src/cupertino/context_menu.dart:777]: Done

**Pull request merged**